### PR TITLE
Add RBAC proxy to metrics

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now use the kube-rbac-proxy to provide two metrics endpoints. Those
get exposed through the `metrics-spod` and `metrics-controller-runtime`
services, serving an authorized self-signed TLS connection.

To be able to retrieve the metrics, it is now required to deploy a
`ClusterRole` and `ClusterRoleBinding` for a `ServiceAccount`. For
example:

```yaml
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: spo-metrics-client
rules:
- nonResourceURLs:
  - /metrics
  verbs:
  - get
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: spo-metrics-client
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: spo-metrics-client
subjects:
- kind: ServiceAccount
  name: default
  namespace: security-profiles-operator
```

The above example is also part of the default deployment, and can be
tested via `kubectl run` within the `security-profiles-operator`
namespace by running:

```
> kubectl run --rm -it pod --image=alpine -- sh -c \
    'apk update && apk add bash curl &&
     curl -vsk -H "Authorization: Bearer `cat /var/run/secrets/kubernetes.io/serviceaccount/token`"
     https://metrics-spod/metrics'
…
*   Trying 10.0.0.222:443...
* Connected to metrics-spod (10.0.0.222) port 443 (#0)
…
> authorization: Bearer xxx
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 200
< content-type: text/plain; version=0.0.4; charset=utf-8
< date: Wed, 28 Apr 2021 12:00:46 GMT
…
security_profiles_operator_seccomp_profile{operation="update"} 1
```

Documentation and e2e testing will follow in further patches.
#### Which issue(s) this PR fixes:

Refers to #422 
#### Does this PR have test?

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Deploying kube-rbac-proxy sidecar in SPOD for exposing metrics via the new `metrics-spod` and `metrics-controller-runtime` services.
```

### CODE REVIEW

File: internal/pkg/manager/spod/bindata/spod.go [hunk 1]
```diff
@@ -352,6 +357,72 @@ semodule -i /opt/spo-profiles/selinuxd.cil
...
+								v1.ResourceEphemeralStorage: resource.MustParse("20Mi"),
+							},
+						},
+						SecurityContext: &v1.SecurityContext{
+							AllowPrivilegeEscalation: &falsely,
+							RunAsUser:                &userRootless,
+							SELinuxOptions: &v1.SELinuxOptions{
+								// TODO(pjbgf): Use a more restricted selinux type
+								Type: "spc_t",
+							},
```
Comment (JAORMX) [inline internal/pkg/manager/spod/bindata/spod.go:387]: This probably doesn't need `spc_t`. Let's leave that empty so it uses the default `container_t`

**Review by JAORMX — CHANGES_REQUESTED** [review]
the rbac-proxy isn't using TLS, is it?

**Comment by saschagrunert** [comment]
> the rbac-proxy isn't using TLS, is it?

It is using a self-signed one right now and we also can provide an external one.

File: internal/pkg/manager/spod/bindata/spod.go (hunk 1 above)
Comment (saschagrunert) [reply internal/pkg/manager/spod/bindata/spod.go]: Removed as suggested :heavy_check_mark:

File: deploy/namespace-operator.yaml [hunk 2]
```diff
@@ -1533,14 +1546,59 @@ kind: Service
...
+    targetPort: 8442
+  selector:
+    app: security-profiles-operator
+    name: spod
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: security-profiles-operator
```
Comment (JAORMX) [inline deploy/namespace-operator.yaml:1564]: Should this point to the label for the SPOD? Seems like the spod has a different label.

File: deploy/base/service.yaml [hunk 3]
```diff
@@ -14,11 +14,25 @@ spec:
...
 spec:
   ports:
-    - port: 80
-      targetPort: 8081
+  - name: https
+    port: 443
+    targetPort: 8442
   selector:
-    app: spod
+    name: spod
```
Comment (JAORMX) [inline deploy/base/service.yaml:25]: should this point to the manager's label? It would be `name: security-profiles-operator`

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: deploy/base/service.yaml (hunk 3 above)
Comment (saschagrunert) [reply deploy/base/service.yaml:25]: No, the metrics endpoints are served by the spod only.

**Review by JAORMX — COMMENTED** [review]
/lgtm

**Pull request merged**